Lib/Database/MySqlQuery.php

```diff
@@ -0,0 +1,1247 @@
+        if (!is_array($clause) || count($clause) < 2 || count($clause) > 3)
+          throw new RuntimeException('Arguments passed to `Query::'.$method.'()` aren\'t in the correct form');
+        $count = 0;
+        foreach ($clause as $scalar) {
+          if (!is_scalar($scalar) && (!is_array($scalar) || $count != 3))
+            throw new RuntimeException('Arguments passed to `Query::'.$method.'()` aren\'t in the correct form');
+          $count++;
+        }
+      }
+
+      // The array is in the correct form.
+      try {
+        $this->buildWhereClause($args[0], $method == 'where' ? 'AND' : 'OR');
+      } catch (Exception $e) {
+        throw $e;
+      }
+    } elseif (is_scalar($args[0])) {
+      // Method was called with a list of arguments.
+      // Check if the arguments are all scalar, and if they are two or three.
+      if (count($args) < 2 || count($args) > 3)
+        throw new RuntimeException('Arguments passed to `Query::'.$method.'()` aren\'t in the correct form');
+        $count = 0;
+      foreach ($args as $clause) {
+        if (!is_scalar($clause) && (!is_array($clause) || $count != 3))
+          throw new RuntimeException('Arguments passed to `Query::'.$method.'()` aren\'t in the correct form');
+        $count++;
+      }
+
+      // Arguments are in the correct form.
+      try {
+        $this->buildWhereClause(array($args), $method == 'where' ? 'AND' : 'OR');
+      } catch (Exception $e) {
+        throw $e;
+      }
+    } else {
+      // Method was called in an incorrect way.
+      throw new RuntimeException('Arguments passed to `Query::'.$method.'()` aren\'t in the correct form');
+    }
+
+    return $this;
+  }
+
+  /**
+   * Process WHERE clauses.
+   *
+   * Process `WHERE ... IN ...` and `WHERE ... NOT IN ...` clauses.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @param string $field Field name.
+   * @param array $values Values to check with.
+   * @param string $method Name of the method that called this one.
+   * @return self
+   *
+   * @throws \RuntimeException if $value elements aren't in the correct form.
+   * @throws SqueezyWeb\Exceptions\InvalidArgumentException if $field isn't a
+   * string.
+   */
+  private function processWhereIn($field, array $values, $method = 'whereIn') {
+    if (!is_string($field))
+      throw InvArgExcp::typeMismatch('field name', $field, 'String');
+
+    $where = $this->where;
+    if ($where != '')
+      $where = 'WHERE '.$field.' ';
+    else
+      $where .= ' AND '.$field.' ';
+    if ($method == 'whereNotIn')
+      $where .= 'NOT ';
+    $where .= 'IN (';
+
+    $count = 0;
+    foreach ($values as $value) {
+      // Correct $value.
+      try {
+        $correct = self::correctValue($value, $method);
+      } catch (Exception $e) {
+        throw $e;
+      }
+
+      // Attach conditions to WHERE clause.
+      if ($count != 0)
+        $where .= ', ';
+      $where .= $correct;
+
+      $count++;
+    }
```

Comment:
This could be improved, for example by using `array_map()`:
```php
$correct =array_map(array(__CLASS__, 'correctValue'), $values, array_fill(0, count($values), $method));
$where .= join(',', $correct);
```